examples/aggregation.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+
+#   Copyright (c) 2010-2014, MIT Probabilistic Computing Project
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Aggregation of probe results over bdb files and model numbers.
+
+A single probe is a typed result of some computation over a BayesDB,
+whose variation with model count, analysis iteration, or other setup
+one is interested in studying.
+
+A probe function is a function that accepts a bdb and computes a list
+of probe results (more than one so that they can share intermediate
+computations).
+
+A probe result is represented as a 3-tuple: the name of the probe, the
+type of the result (as a string), and the value of the result.
+
+Two result types are currently supported:
+- 'num' results are aggregated by maintaining a list of all of them.
+- 'bool' results are aggregated by maintaining the count of True and
+  False results seen.
+
+"""
+
+import contextlib
+import random
+import time
+
+import bayeslite
+from bayeslite.core import bayesdb_get_generator
+from bayeslite.metamodels.crosscat import CrosscatMetamodel
+from crosscat.LocalEngine import LocalEngine as CrosscatLocalEngine
+
+then = time.time()
```

Comment:
General habit: I usually name the current time `now`, so the difference becomes `now - then`.  In this case, though, since it's only used in one place, fixed in e5c012149628c85daf4abd3eb0e8b42bd7ac4537